sql/analyzer/triggers.go

```diff
@@ -189,43 +194,45 @@ func applyTriggers(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope) (sql
 	}
 
 	if len(affectedTriggers) == 0 {
-		return n, nil
+		return n, transform.SameTree, nil
 	}
 
 	triggers := orderTriggersAndReverseAfter(affectedTriggers)
 	originalNode := n
-
+	same := transform.SameTree
+	allSame := transform.SameTree
 	for _, trigger := range triggers {
 		err = validateNoCircularUpdates(trigger, originalNode, scope)
 		if err != nil {
-			return nil, err
+			return nil, transform.SameTree, err
 		}
 
-		n, err = applyTrigger(ctx, a, originalNode, n, scope, trigger)
+		n, same, err = applyTrigger(ctx, a, originalNode, n, scope, trigger)
 		if err != nil {
-			return nil, err
+			return nil, transform.SameTree, err
 		}
+		allSame = same && allSame
 	}
 
-	return n, nil
+	return n, allSame, nil
 }
 
 // applyTrigger applies the trigger given to the node given, returning the resulting node
-func applyTrigger(ctx *sql.Context, a *Analyzer, originalNode, n sql.Node, scope *Scope, trigger *plan.CreateTrigger) (sql.Node, error) {
+func applyTrigger(ctx *sql.Context, a *Analyzer, originalNode, n sql.Node, scope *Scope, trigger *plan.CreateTrigger) (sql.Node, transform.TreeIdentity, error) {
```

Comment:
I think by definition this always either alters the tree or returns an err

Reply:
is it possible for one of the recursive nodes to have a createTrigger inside?